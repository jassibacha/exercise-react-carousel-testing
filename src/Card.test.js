import React from 'react';
import { render } from '@testing-library/react';
import Card from './Card';

// Smoke Test
it('renders without crashing', function () {
    render(
        <Card caption="Test Caption" src="test.jpg" currNum={1} totalNum={5} />
    );
});

// Snapshot Test
it('matches snapshot', function () {
    const { asFragment } = render(
        <Card caption="Test Caption" src="test.jpg" currNum={1} totalNum={5} />
    );
    expect(asFragment()).toMatchSnapshot();
});
