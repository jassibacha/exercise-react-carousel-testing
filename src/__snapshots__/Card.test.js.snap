// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Test Caption
    </h4>
    <img
      alt="Test Caption"
      class="Card-image"
      src="test.jpg"
    />
    <small
      class="Card-small"
    >
      Image 1 of 5.
    </small>
  </div>
</DocumentFragment>
`;
